### PR TITLE
Add DEBUGPY_TRACE_DEBUGPY variable to allow debugpy to debug itself

### PR DESCRIPTION
**Pull request by rchiodo**
To allow using debugpy to debug itself (well at least the adapter), added a new environment variable that skips setting `dont_trace_start_patterns` and setting thread flags that hide threads.

### CODE REVIEW

File: src/debugpy/common/log.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def __init__(self, filename, file, levels=LEVELS, close_file=True):
                 platform.machine(),
                 platform.python_implementation(),
                 platform.python_version(),
-                64 if sys.maxsize > 2**32 else 32,
```
Comment (rchiodo) [inline src/debugpy/common/log.py:61]: THis is a 'black' change. I just ran black on everything and it added this.

File: src/debugpy/launcher/handlers.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 from debugpy.common import json
 from debugpy.launcher import debuggee
 
+
```
Comment (rchiodo) [inline src/debugpy/launcher/handlers.py:13]: another black change

File: src/debugpy/common/util.py [hunk 3]
```diff
@@ -148,3 +148,18 @@ def srcnameof(obj):
...
+
+
+def disable_tracing(thread):
+    """Disables tracing for the given thread if DEBUGPY_TRACE_DEBUGPY is not set.
+    DEBUGPY_TRACE_DEBUGPY is used to debug debugpy with debugpy
+    """
+    if skip_trace():
+        thread.pydev_do_not_trace = True
+        thread.is_pydev_daemon_thread = True
+        thread.daemon = True
```
Comment (int19h) [inline src/debugpy/common/util.py:165]: This one doesn't belong here - it affects the behavior of the thread when the process shuts down (whether it waits for this thread or not before exiting), not whether it's traced by the debugger.

File: src/debugpy/common/util.py [hunk 4]
```diff
@@ -148,3 +148,18 @@ def srcnameof(obj):
...
 
     return name
+
+
+def skip_trace():
+    """Returns True if tracing should be skipped for the current thread."""
+    return "DEBUGPY_TRACE_DEBUGPY" not in os.environ
+
+
+def disable_tracing(thread):
```
Comment (int19h) [inline src/debugpy/common/util.py:158]: Let's call this something like `hide_thread_from_debugger` to distinguish it from other uses of "tracing" (that pertain to the debuggee) elsewhere in the code. 

Similarly for `skip_trace` - something like `hide_debugpy_internals` would be clearer, and will cover any other future tweaks we need to do to keep this working.

Comment (rchiodo) [reply src/debugpy/common/util.py:158]: Sounds good, will do.

File: src/debugpy/common/util.py (hunk 3 above)
Comment (rchiodo) [reply src/debugpy/common/util.py:165]: Okay, wasn't sure. I'll put it back.

**Review by int19h — APPROVED** [review]

**Comment by rchiodo** [comment]
I'm guessing the linux 3.9 test is flakey? It's checking thread count, so I thought it might be related.

**Pull request merged**